packages/gnucobol/build.sh

```diff
@@ -2,20 +2,31 @@ TERMUX_PKG_HOMEPAGE=https://gnucobol.sourceforge.io/
 TERMUX_PKG_DESCRIPTION="A free/libre COBOL compiler"
 TERMUX_PKG_LICENSE="GPL-3.0, LGPL-3.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION=3.1.2
-TERMUX_PKG_REVISION=3
-TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/gnucobol/gnucobol-${TERMUX_PKG_VERSION}.tar.xz
-TERMUX_PKG_SHA256=597005d71fd7d65b90cbe42bbfecd5a9ec0445388639404662e70d53ddf22574
-TERMUX_PKG_DEPENDS="json-c, libgmp, libvbisam, libxml2, ncurses"
+TERMUX_PKG_VERSION=3.2
+TERMUX_PKG_REVISION=0
+
+TERMUX_PKG_SRCURL=git+https://github.com/alexbodn/gnucobol-3.2.git
```

Comment:
Do you really need to change the upstream tarball?
Note that as-is you very likely trash the timestamps in that way, forcing a recreation of many files which wouldn't be necessary normally.

Reply:
it was your suggestion to put the last release.
but i'm serving it off my github until you'd fix the issues upstream.

Reply:
and the termux people didn't like the huge diff created by configure change.

Reply:
> and the termux people didn't like the huge diff created by configure change.

My main issue with it is that it is unmaintainable.
It's gonna stop applying and need triage as soon as there is any update to the package.

Reply:
I should take this opportunity to note that changing the source URL to your personal fork is a complete no go unless you have a good explanation for why that is necessary.

Reply:
> I should take this opportunity to note that changing the source URL to your personal fork is a complete no go unless you have a good explanation for why that is necessary.

you rejected my previous port that applied a big patch as unmaintainable.
that patch was not termux related indeed and I offered it to gnucobol creators upstream.
untill they'll provide this functionality officially I'm serving it myself, but will happily adjust as they go.
as an alternative, I found a way to submit the upstream patch in a lower signature.

Reply:
> > and the termux people didn't like the huge diff created by configure change.
> 
> My main issue with it is that it is unmaintainable. It's gonna stop applying and need triage as soon as there is any update to the package.

one patch files adjust some /tmp literal to termux,  and the other prevents make from running some programs when cross compiling, as it happens for termux. @GitMensch had another suggestion for the second one.

Reply:
Pulling in pending patches is fine as a second entry in `TERMUX_PKG_SRCURL`.
I know we've done that before for but I can't for the life of me find that PR.
I do recall it involved @Biswa96, and upstream merging a PR within like 5 minutes of it getting brought up in our PR.

Reply:
Ah found it.
- https://github.com/termux/termux-packages/pull/19587#discussion_r1559514682

Reply:
That is something else - the upstream package was accepted, but that doesn't mean that there is a stable release with that. If you use the current 3.3 snapshot then there are a bunch of patches already included (for example the inclusion for xml2 parser.h) as well as there will be configure adjustments for BDB and cross-compilation.
But I think you'd like to have a stable release + patches, no?

Back to common.c: is there _really_ a need to adjust the code that has "/tmp" hardcoded, when it is only a fallback for the case that none of TMPDIR, TMP, TEMP environment variables are set?

Reply:
Can drop the /tmp patch. If upstream guarantees it works, Termux can gladly accept.

Reply:
We can guarantee `$TMPDIR` to have a consistent value at build time.
We can't guarantee that users won't willfully break their environments.

Reply:
In this case I'd say let's give them the fun to have a broken GnuCOBOL that way and drop the patch.

Reply:
> > and the termux people didn't like the huge diff created by configure change.
> 
> My main issue with it is that it is unmaintainable. It's gonna stop applying and need triage as soon as there is any update to the package.

I may argue, that each package is targeting a specific version of upstream, with a very specific sha256sum.

i had to change things when switching from gnucobol 3.1.2 to 3.2, but had to extend the changes since important functionality failed in termux, initially with both upstream versions btw.

it's the same with debian packages.

Reply:
> I may argue, that each package is targeting a specific version of upstream, with a very specific sha256sum.

The `TERMUX_PKG_SHA256` field is a security and integrity measure.
It helps ensure that package sources have not been tampered with or corrupted.
The binding to a specific version is a result of this, though somewhat incidental.

Reply:
sure.
they don't have a patched version yet.
I've tried to upload to termux the patch to upstream, and it was ugly and unrelated somehow.

> In this case I'd say let's give them the fun to have a broken GnuCOBOL that way and drop the patch.

I think it's a pitty.
if upstream gives this fallback option, why not on termux too?
it's definitely and legitimately termux related.